src/server/scripts/EasternKingdoms/ZulGurub/boss_venoxis.cpp

```diff
@@ -80,275 +65,213 @@ enum Phases
-                        break;
-                    case EVENT_HOLY_NOVA:
-                        DoCastSelf(SPELL_HOLY_NOVA);
-                        events.ScheduleEvent(EVENT_HOLY_NOVA, 10s,  24s, 0, PHASE_ONE);
-                        break;
-
-                    //
-                    // snake form spells and Actions
-                    //
-
-                    case EVENT_VENOM_SPIT:
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random))
-                            DoCast(target, SPELL_VENOM_SPIT);
-                        events.ScheduleEvent(EVENT_VENOM_SPIT, 5s, 15s, 0, PHASE_TWO);
-                        break;
-                    case EVENT_POISON_CLOUD:
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random))
-                            DoCast(target, SPELL_POISON_CLOUD);
-                        events.ScheduleEvent(EVENT_POISON_CLOUD, 15s, 20s, 0, PHASE_TWO);
-                        break;
-                    case EVENT_PARASITIC_SERPENT:
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random))
-                            DoCast(target, SPELL_SUMMON_PARASITIC_SERPENT);
-                        events.ScheduleEvent(EVENT_PARASITIC_SERPENT, 15s, 0, PHASE_TWO);
-                        break;
-                    case EVENT_FRENZY:
-                        // frenzy at 20% health
-                        DoCast(me, SPELL_FRENZY, true);
-                        break;
-
-                    //
-                    // shape and phase-changing
-                    //
-
-                    case EVENT_TRANSFORM:
-                        // shapeshift at 50% health
-                        DoCast(me, SPELL_VENOXIS_TRANSFORM);
-                        Talk(SAY_VENOXIS_TRANSFORM);
-                        DoResetThreatList();
-
-                        // phase two events (snakeform)
-                        events.ScheduleEvent(EVENT_VENOM_SPIT, 5s, 0, PHASE_TWO);
-                        events.ScheduleEvent(EVENT_POISON_CLOUD, 10s, 0, PHASE_TWO);
-                        events.ScheduleEvent(EVENT_PARASITIC_SERPENT, 30s, 0, PHASE_TWO);
-
-                        // transformed, start phase two
-                        events.SetPhase(PHASE_TWO);
-
-                        break;
-                    default:
-                        break;
-                }
-            }
-
-            DoMeleeAttackIfReady();
-        }
+                DoCastSelf(SPELL_FRENZY, true);
+            });
+        });
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        BossAI::JustDied(killer);
+        Talk(SAY_VENOXIS_DEATH);
+        me->RemoveAllAuras();       // removes transform
+    }
 
-    private:
-        uint8 _inMeleeRange;
-        bool _transformed;
-        bool _frenzied;
-    };
+private:
+    // Spawns the Razzashi Cobra adds before the encounter starts
+    void _spawnCobras()
```

Comment:
Consider pre spawning them and making them minions. See magtheridon and fathomlord for an example. You need to alter instance_zulgurub for it

Reply:
A new system to learn! There's always one more...

Just for my education, what is the functional difference between tracking them as minions and tracking them as summons?

Reply:
Mostly that summoning creatures normally is not really blizzlike. In wotlk classic there is strong evidence that at least for boss fights all potential summons are done with spells. Other than that it is useful to prespawn them and put them in the creature table so they can be more easily affected by instance related effects. Plus it is more elegant and cleans up the code imo